src/components/AutoUpdateTime.js

```diff
@@ -22,85 +27,70 @@ const propTypes = {
     ...withLocalizePropTypes,
 };
 
-class AutoUpdateTime extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.getCurrentUserLocalTime = this.getCurrentUserLocalTime.bind(this);
-        this.updateCurrentTime = this.updateCurrentTime.bind(this);
-        this.getTimezoneName = this.getTimezoneName.bind(this);
-        this.state = {
-            currentUserLocalTime: this.getCurrentUserLocalTime(),
-        };
-    }
-
-    componentDidMount() {
-        this.updateCurrentTime();
-    }
-
-    componentDidUpdate() {
-        // Make sure the interval is up to date every time the component updates
-        this.updateCurrentTime();
-    }
-
-    componentWillUnmount() {
-        clearTimeout(this.timer);
-    }
-
+function AutoUpdateTime(props) {
     /**
      * @returns {moment} Returns the locale moment object
      */
-    getCurrentUserLocalTime() {
-        return DateUtils.getLocalMomentFromDatetime(
-            this.props.preferredLocale,
+    const getCurrentUserLocalTime = useCallback(() => (
+        DateUtils.getLocalMomentFromDatetime(
+            props.preferredLocale,
             null,
-            this.props.timezone.selected,
-        );
-    }
+            props.timezone.selected,
+        )
+    ), [props.preferredLocale, props.timezone.selected]);
+
+    const [currentUserLocalTime, setCurrentUserLocalTime] = useState(getCurrentUserLocalTime());
+    const timerRef = useRef(null);
 
     /**
      * @returns {string} Returns the timezone name in string, e.g.: GMT +07
      */
-    getTimezoneName() {
+    const getTimezoneName = useCallback(() => {
```

Comment:
NAB, but I'd maybe reach for `useMemo()` instead here since we're only using this in the JSX.

It'd basically be the same but we'd change `getTimezoneName` to something like:

```js
const timezoneName = useMemo(() => {...}, [currentUserLocalTime]);
```

Then use it directly instead of calling a function.

Though they are basically equivalent.

Reply:
Oh cool, that makes sense.